waspc/src/Wasp/Generator/DbGenerator/Common.hs

```diff
@@ -93,21 +90,8 @@ dbSchemaChecksumOnLastGenerateFileInDbRootDir = [relfile|schema.prisma.wasp-gene
 dbSchemaChecksumOnLastGenerateFileProjectRootDir :: Path' (Rel ProjectRootDir) (File DbSchemaChecksumOnLastGenerateFile)
 dbSchemaChecksumOnLastGenerateFileProjectRootDir = dbRootDirInProjectRootDir </> dbSchemaChecksumOnLastGenerateFileInDbRootDir
 
-prismaClientOutputDirEnvVar :: String
-prismaClientOutputDirEnvVar = "PRISMA_CLIENT_OUTPUT_DIR"
-
-prismaClientOutputDirInAppComponentDir :: AppComponentRootDir d => Path' (Rel d) (Dir ServerRootDir)
-prismaClientOutputDirInAppComponentDir = [reldir|node_modules/.prisma/client|]
-
-serverPrismaClientOutputDirEnv :: (String, String)
-serverPrismaClientOutputDirEnv = appComponentPrismaClientOutputDirEnv serverRootDirFromDbRootDir
-
-webAppPrismaClientOutputDirEnv :: (String, String)
-webAppPrismaClientOutputDirEnv = appComponentPrismaClientOutputDirEnv webAppRootDirFromDbRootDir
-
-appComponentPrismaClientOutputDirEnv :: AppComponentRootDir d => Path' (Rel DbRootDir) (Dir d) -> (String, String)
-appComponentPrismaClientOutputDirEnv appComponentDirFromDbRootDir =
-  (prismaClientOutputDirEnvVar, SP.fromRelDir $ appComponentDirFromDbRootDir </> prismaClientOutputDirInAppComponentDir)
+waspProjectDirFromProjectRootDir :: Path' (Rel ProjectRootDir) (Dir WaspProjectDir)
+waspProjectDirFromProjectRootDir = [reldir|../../|]
```

Comment:
Ok this is maybe a bit weird hm. You are going in the opposite direction, trying to get to user's code from the generated code dir, right? (terrible names I know: waspProjectDir and projectRootDir).

Since we certainly have information about path from waspProjectDir to projectRootDir, can't you use that info to construct this path? You should be able to "reverse" that path. And then this one gets defined via that one and won't become incorrect.

We have function `reversePosixPath`, look it up -> we could use something like that? I guess the problem is it works only for Posix.

Poor man's solution is always to put  a comment here that this depends on this and that variable, and then also go to those variables and put a comment those variables that they are connected with this one, so when somebody changes on of these, they also have to change the others.

Reply:
I think I now remembered -> thi information is not hardcoded in a variable but is actually being passed as an argument? In that case, we either need to hardcode it as top level variable, or we will want to actually pass it around properly.

Reply:
This is the dirty part I was referring to earlier. Let's discuss in in the other thread.

Just to repeat some of the points explained in the other thread:

> or we will want to actually pass it around properly.

This is what I've tried to avoid. Relevant comments: https://github.com/wasp-lang/wasp/pull/1604#discussion_r1431304174 and https://github.com/wasp-lang/wasp/pull/1604#discussion_r1437745002.

> we either need to hardcode it as top level variable

Unfortunately, this isn't possible because `ProjectRootDir` can be either `out` or `build`.

Reply:
And I just suggested in that comment thread that we continue the discussion here :D.

Even though it can be either `out` or `build`, we can still hardcode it -> it just means that the relative path is the same regardless of which one of these is it.